src/Engine/Renderer/Texture/TextureManager.hpp

```diff
@@ -35,32 +38,67 @@ struct TextureData {
     void* data = nullptr;
 };
 
+/**
+ * Manage Texture loading and registration.
+ */
 class RA_ENGINE_API TextureManager final {
     RA_SINGLETON_INTERFACE( TextureManager );
 
   private:
     using TexturePair = std::pair<std::string, Texture*>;
 
   public:
+    /** Add a texture giving its name, dimension and content.
+     * Usefull for defining procedural textures
+     */
     TextureData& addTexture( const std::string& name, int width, int height, void* data );
-    Texture* addTexture( const std::string& filename );
+
+    /**
+     * Get or load texture from a file.
+     * The name of the texture is the name of its file
+     * @param filename
+     * @return
+     */
     Texture* getOrLoadTexture( const std::string& filename );
+
+    /**
+     * Get or load a named texture.
+     * The name of the texture might be different of the associated file but the data must be
+     * loaded in the TextureData before calling this method.
+     * @param filename
+     * @return
+     */
     Texture* getOrLoadTexture( const TextureData& data );
 
+    /**
+     * Delete a named texture from the manager
+     * @param filename
+     */
     void deleteTexture( const std::string& filename );
+    /**
+     * Delete a texture from the manager
+     * @param texture
+     */
     void deleteTexture( Texture* texture );
 
-    // Call this method to update given texture
+    /// Call this method to update given texture
     void updateTexture( const std::string& texture, void* data );

```

Comment:
Can't the type of data be deduced from the texture properties ?
It could be nice to avoid this `void*`.

Reply:
Well, not really ... at least, not in a simple way ...

Reply:
ok no problem.